lib/main_common.dart

```diff
@@ -87,9 +90,30 @@ Future<void> _firebaseInitializer(FlavorName flavorName) async {
   await Firebase.initializeApp(options: firebaseOption);
 }
 
+
 void _activateManagers() {
-  activateRemoteConfig();
+  getConnectionStatus();
+
   if (!kIsWeb) {
     activateCrashReportIfEnabled();
   }
 }
+
+void getConnectionStatus() {
+  final ConnectionStatus connectionStatus =
+      DependecyInjection.instance.get<ConnectionStatus>();
+
+  connectionStatus.initialize();
+  //Listen for connection change
+  connectionStatus.connectionChangeStream
+      .asBroadcastStream()
+      .listen(connectionChanged);
+}
+
+void connectionChanged(dynamic hasConnection) {
+  bool hasInterNetConnection = false;
+  hasInterNetConnection = hasConnection;
+  if (hasInterNetConnection) {
+    activateRemoteConfig();
+  }
+}
```

Comment:
Move these methods from the main file. 
Put them directly in the manager file and just import the file to use the method